src/networking.c

```diff
@@ -2949,7 +2949,7 @@ NULL
         if (getLongLongFromObjectOrReply(c,c->argv[2],&id,NULL)
             != C_OK) return;
         struct client *target = lookupClientByID(id);
-        if (target && target->flags & CLIENT_BLOCKED && moduleBlockedClientMayTimeout(target)) {
+        if (target && target->flags & CLIENT_BLOCKED && (unblock_error || moduleBlockedClientMayTimeout(target))) {
```

Comment:
didn't we have some issue with module clients with no timeout callback that can't be unblocked (#9167)?

Reply:
good catch, but i raise you this:

c1:
```
127.0.0.1:6379> fsl.bpop k 0
```

c2:
```
127.0.0.1:6379> slaveof localhost 5555
```

c1:
```
127.0.0.1:6379> fsl.bpop k 0
(error) UNBLOCKED force unblock from blocking operation, instance state changed (master -> replica?)
```

so even if a client does not expect to be forcefully unblocked (i.e. no timeout callback), the situation still exists.

@yossigo thoughts?

Reply:
@guybe7 This form of forced unblock also ends up disconnecting the client. From the perspective of a module that did not provide a timeout callback, I think this should be handled exactly the same a client that dropped while it was blocked.

Reply:
ok, i can't say i agree 100% but i don't feel too strongly about it.
reverted.